src/lib/Parser.hs

```diff
@@ -258,7 +258,7 @@ findImplicitImplicitArgNames typ = filter isLowerCaseName $ envNames $
     UFor _ _ _ -> error "Unexpected for in type annotation"
     UHole -> mempty
     UTypeAnn v ty -> findVarsInAppLHS v <> findVarsInAppLHS ty
-    UTabCon _ -> error "Unexpected table in type annotation"
+    UTabCon _ -> mempty
```

Comment:
Why did that have to change? It looks wrong to me

Reply:
This is reachable with a syntactically valid but ill-typed user program so it shouldn't cause a compiler crash. We could throw a parser error (and change the type of this function to allow that) but I think it's also fine to forget about the implicit args and catch the error during type inference.

Reply:
Right. So the whole explanation sounds reasonable to me, but:
1. This should be explained in the comment
2. There are more error cases that we should fix
Also, do we really have a _guarantee_ that there will be a type error if there's a `UTabCon`? At some point we might add another syntactic form that turns a `UTabCon` into a valid type, and then this function will silently compute the wrong thing.

Reply:
Sure. I'll fix the rest of the function too.